packages/react-button/src/components/Button/createButton.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { mergeProps, getSlots, simplifyShorthand } from '@fluentui/react-compose/lib/next/index';
+import { ButtonProps, ButtonState } from './Button.types';
+import { useButton } from './useButton';
+
+/**
+ * Consts listing which props are shorthand props.
+ */
+export const buttonShorthandProps = ['icon', 'loader', 'children'];
+
+/**
+ * Define the render function. Given the state of a button, renders it.
+ */
+export const renderButton = (state: ButtonState) => {
+  const { slots, slotProps } = getSlots(state, buttonShorthandProps);
+  const { loading, iconPosition, iconOnly } = state;
+
+  return (
+    <slots.root {...slotProps.root}>
+      {loading && <slots.loader {...slotProps.loader} />}
+      {iconPosition !== 'after' && <slots.icon {...slotProps.icon} />}
+      {!iconOnly && <slots.children {...slotProps.children} />}
+      {iconPosition === 'after' && <slots.icon {...slotProps.icon} />}
+    </slots.root>
+  );
+};
+
+/**
+ * Given user props, returns state and render function for a Button.
+ */
+export const createButton = (props: ButtonProps, ref: React.Ref<HTMLElement>, defaultProps?: ButtonProps) => {
```

Comment:
this is a hook, we should call it `useButton`

Reply:
This is a factory function, but yes, maybe you're right.

`useButtonFactory`?

Reply:
then we wouldn't have to disable the rule :)

Reply:
I would say that this is the `useButton` hook. The other one should either be called: `useButtonAria` if it handles only a11y, or `useButtonState`.

Reply:
alright i'll go with that.